common/types/activation_test.go

```diff
@@ -36,3 +41,22 @@ func unsafeMul(x, y uint64) uint64 {
 	result = x // avoid compiler optimizations
 	return x * y
 }
+
+func layerTester(tb testing.TB, encodable codec.Encodable) LayerID {
+	tb.Helper()
+	f := fuzz.NewWithSeed(1001)
+	f.Fuzz(encodable)
+	buf, err := codec.Encode(encodable)
+	require.NoError(tb, err)
+	var lid LayerID
+	require.NoError(tb, codec.Decode(buf, &lid))
+	return lid
+}
+
+func TestActivationEncoding(t *testing.T) {
+	t.Run("layer is first", func(t *testing.T) {
+		atx := ActivationTx{}
+		lid := layerTester(t, &atx)
+		require.Equal(t, atx.PubLayerID, lid)
```

Comment:
IMO, the `atx.PubLayerID` should be set to be certain:
```suggestion
		atx := ActivationTx{PubLayerID: LayerID{Value: 123}}
		lid := layerTester(t, &atx)
		require.Equal(t, atx.PubLayerID, lid)
```

Same in other tests